src/node.rs

```diff
@@ -142,9 +146,48 @@ impl<S: StorageExt> SawtoothRaftNode<S> {
             },
             _ => false,
         } {
-            debug!("Leader({:?}) transition to Building block {:?}", self.peer_id, block_id);
-            self.leader_state = Some(LeaderState::Building(Instant::now()));
-            self.service.initialize_block(None).expect("Failed to initialize block");
+            // Get list of peers from settings
+            let settings = self.service
+                .get_settings(block_id.clone(), vec![String::from("sawtooth.consensus.raft.peers")])
+                .expect("Failed to get settings");
+            let peers = get_peers_from_settings(&settings);
+            let peers: HashSet<PeerId> = HashSet::from_iter(peers);
+
+            // Check if membership has changed
+            let old_peers: HashSet<PeerId> = HashSet::from_iter(self.raft_id_to_peer_id.values().cloned());
+            let difference: HashSet<PeerId> = peers.symmetric_difference(&old_peers).cloned().collect();
+
+            if !difference.is_empty() {
+                if difference.len() > 1 {
+                    panic!("More than one node's membership has changed; only one change can be processed at a time.");
+                }
+
+                // The raft-rs library can only add or delete one node at a time, so there should
+                // only be one item in this iterator
+                difference.iter().for_each(|peer_id| {
+                    // Initialize change
+                    let mut change = ConfChange::new();
+                    change.set_node_id(peer_id_to_raft_id(&peer_id));
+
+                    // Check if adding or removing Node
+                    if peers.len() > old_peers.len() {
+                        change.set_change_type(ConfChangeType::AddNode);
+                        // Include peer ID in context so node can be added to raft_id_to_peer_id map later
+                        change.set_context(Vec::from(peer_id.clone()));
+                    } else if peers.len() < old_peers.len() {
+                        change.set_change_type(ConfChangeType::RemoveNode);
+                    }
+
+                    info!("Leader({:?}) detected configuration change {:?}", self.peer_id, change);
+                    debug!("Leader({:?}) transition to ChangingConfig", self.peer_id);
+                    self.leader_state = Some(LeaderState::ChangingConfig);
+                    self.raw_node.propose_conf_change(vec![], change).unwrap();
+                });
+            } else {
+                debug!("Leader({:?}) transition to Building block {:?}", self.peer_id, block_id);
+                self.leader_state = Some(LeaderState::Building(Instant::now()));
+                self.service.initialize_block(None).expect("Failed to initialize block");
+            }
```

Comment:
This change should be refactored behind a private method. Something like `check_for_conf_change(&self) -> Option<ConfChange>` would probably be clearer and a good level of abstraction.